backend/src/routes/transactions.py

```
@@ -95,3 +109,82 @@ def get_account_transactions(
+        ).all()
+
+        return {"data": transactions, "total_pages": total_pages, "page": page}
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception(e)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An error occurred while fetching transactions",
+        )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)